src/main/java/com/example/gamelist/GameService.java

```diff
@@ -0,0 +1,28 @@
+package com.example.gamelist;
+
+import org.springframework.dao.DuplicateKeyException;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+public class GameService {
+
+    private final GameMapper gameMapper;
+
+    public GameService(GameMapper gameMapper) {
+        this.gameMapper = gameMapper;
+    }
+
+    public void updateGame(final Integer id, String name) {
+        Game game = gameMapper.findById(id).orElseThrow(() -> new GameNotFoundException("game information not found"));
+        List<Game> otherGames = this.gameMapper.findByName(name);
+
+        if (otherGames.stream().anyMatch(game -> game.getName().equals(name))) {
```

Comment:
gameだと変数名が被ってしまうので、 g という変数名にしましょうか。
原則 g などアルファベット一文字の変数名は中身がわかりづらくアンチパターンですがラムダにおいては g が何を表すか明白なので特別にアルファベット一文字でも良いです。
```suggestion
        if (otherGames.stream().anyMatch(g -> g.getName().equals(name))) {
```

Reply:
2ab9a159c82163a6398c9fb1b0eea212e80dbdcb
<img width="696" alt="スクリーンショット 2024-01-02 16 34 48" src="https://github.com/otyacazyu/gamelist/assets/143002834/8f1fc57b-45ca-4c55-bbc4-e41d9a3e6b60">

編集を行いました。

Reply:
OKです。